### PR TITLE
acceleration to cluster changes

### PR DESCRIPTION
**Pull request by chancity**
## Description

acceleration to cluster changes
## Motivation and Context

if cluster distance is <= X then speed

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: bot.user.js [hunk 1]
```diff
@@ -595,10 +595,14 @@ var bot = (function() {
                     window.goalCoordinates = coordinatesOfClosestFood;
                     // Sprint
                     //use speed to go to larger clusters
-                    setAcceleration((window.currentFoodScore >= accelerationClusterSize) ? 1 : 0);
+					setAcceleration((window.currentFood.clusterScore >= accelerationClusterSize) 
+					?
+					(canvas.getDistance(window.getX(),window.getY(),window.currentFoodX,window.currentFoodY) <= Math.pow(window.getSnakeLength(), 2) / 2 && window.currentFood.distance > 375) 
+					? 
+					1 : 0 : 0);
```
Comment (ChadSki) [inline bot.user.js:602]: I'm having a hard time unwrapping these nested ternary operators... can you help?

```
if (window.currentFood.clusterScore >= accelerationClusterSize) {
    var distance = canvas.getDistance(window.getX(), window.getY(),
                                      window.currentFoodX, window.currentFoodY);
    var arbitraryCutoff = Math.pow(window.getSnakeLength(), 2) / 2;
    if (distance <= arbitraryCutoff && window.currentFood.distance > 375) {
        setAcceleration(1);
    }
    else {
        setAcceleration(0);
    }
}
else {
    setAcceleration(0);
}
```

**Comment by chancity** [comment]
@ChadSki 

```
distance = canvas.getDistance(window.getX(),window.getY(),window.currentFoodX,window.currentFoodY);
lengthToAccelerate = Math.pow(window.getSnakeLength(), 2) / 2;
isSnakeLongEnoughToAccelerate = (distance <= LengthToAccelerate);
isFoodFarEnoughAwayToAccelerate = (window.currentFood.distance > 375);
isClusterBigEnoughToAccelerate = (window.currentFood.clusterScore >= accelerationClusterSize);

setAcceleration(isClusterBigEnoughToAccelerate && isSnakeLongEnoughToAccelerate && isFoodFarEnoughAwayToAccelerate);
```

**Comment by ChadSki** [comment]
@chancity can you commit it as such?

**Comment by ermiyaeskandary** [comment]
Approved :)

[Approved with PullApprove](https://pullapprove.com/ErmiyaEskandary/Slither.io-bot/pull-request/128/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ErmiyaEskandary/Slither.io-bot)

**Comment by ChadSki** [comment]
I disapprove of nested ternary operators. :(

**Comment by ermiyaeskandary** [comment]
Ah - so it's this commit ? @ChadSki you can revert if you can . If you can't , I'll do it soon.

**Comment by ChadSki** [comment]
I'm away from my home computer for another eight hours or so. I can fix it then if you haven't.

**Pull request merged**